src/Core/Auth/UserFeatures/WebAuthnLogin/Implementations/AssertWebAuthnLoginCredentialCommand.cs

```diff
@@ -46,8 +46,17 @@ public async Task<(User, WebAuthnCredential)> AssertWebAuthnLoginCredential(Asse
         // Always return true, since we've already filtered the credentials after user id
         IsUserHandleOwnerOfCredentialIdAsync callback = (args, cancellationToken) => Task.FromResult(true);
         var credentialPublicKey = CoreHelpers.Base64UrlDecode(credential.PublicKey);
-        var assertionVerificationResult = await _fido2.MakeAssertionAsync(
-            assertionResponse, options, credentialPublicKey, (uint)credential.Counter, callback);
+
+        Fido2NetLib.Objects.AssertionVerificationResult assertionVerificationResult = null;
+        try
+        {
+            assertionVerificationResult = await _fido2.MakeAssertionAsync(
+                assertionResponse, options, credentialPublicKey, (uint)credential.Counter, callback);
+        }
+        catch (Fido2VerificationException)
+        {
+            throw new BadRequestException("Unable to verify credential.");
+        }
```

Comment:
Did I write this?? I must've completely missed the fact that `_fido2.MakeAssertionAsync` can throw exceptions... Good catch!

❓ **issue:** So one thing we need to be careful with here is that we don't allow credential and/or user enumeration. If we have unique exception messages here then the caller will be able to differentiate between 

- `Invalid Credential -> Not found in database`
- `Unable to verify credential -> Found in database, but invalid assertion`

Reply:
I've adjusted the messages with [`bd46adc` (#3873)](https://github.com/bitwarden/server/pull/3873/commits/bd46adce7727750045f31dda692f0caf6d5b50e2).